package/gluon-mesh-batman-adv/src/respondd.c

```diff
@@ -78,47 +78,30 @@ struct clients_netlink_opts {
 
 
 static struct json_object * get_addresses(void) {
-	FILE *f = fopen("/proc/net/if_inet6", "r");
-	if (!f)
-		return NULL;
+	struct ifaddrs *if_as, *if_a;
+	struct json_object *ret;
 
-	char *line = NULL;
-	size_t len = 0;
+	if (getifaddrs(&if_as))
+		return NULL;
 
-	struct json_object *ret = json_object_new_array();
+	ret =  json_object_new_array();
 
-	while (getline(&line, &len, f) >= 0) {
-		/* IF_NAMESIZE would be enough, but adding 1 here is simpler than subtracting 1 in the format string */
-		char ifname[IF_NAMESIZE+1];
-		unsigned int flags;
-		struct in6_addr addr;
+	for (if_a = if_as; if_a; if_a = if_a->ifa_next) {
+		struct sockaddr_in6 *sa_in6;
 		char buf[INET6_ADDRSTRLEN];
 
-		if (sscanf(line,
-			   "%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8
-			   "%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8"%2"SCNx8
-			   "  %*x %*x %*x %x %"STRINGIFY(IF_NAMESIZE)"s",
-			   &addr.s6_addr[0], &addr.s6_addr[1], &addr.s6_addr[2], &addr.s6_addr[3],
-			   &addr.s6_addr[4], &addr.s6_addr[5], &addr.s6_addr[6], &addr.s6_addr[7],
-			   &addr.s6_addr[8], &addr.s6_addr[9], &addr.s6_addr[10], &addr.s6_addr[11],
-			   &addr.s6_addr[12], &addr.s6_addr[13], &addr.s6_addr[14], &addr.s6_addr[15],
-			   &flags, ifname) != 18)
-			continue;
-
-		if (strcmp(ifname, "br-client"))
+		if (!if_a->ifa_addr || if_a->ifa_addr->sa_family != AF_INET6)
 			continue;
 
-		if (flags & (IFA_F_TENTATIVE|IFA_F_DEPRECATED))
```

Comment:
This flag check should not be removed (there's `if_a->ifa_flags`).

Reply:
This gives the interface flag. It seems there is no way to get the address flags by the current approach